int32 x
int32 y
float32 theta
int32 length
int32 width
string color